### ISSUE TITLE
Support LoRA training

### ISSUE THREAD

**Issue by taha-yassine** [OpenHands/agentic-code-search-oss#53]

**Comment by ApGa** [comment]
BTW, tinker-cookbook has a utility to estimate an appropriate learning rate to use with LoRa (currently they just multiply full-finetuning lr by 10): https://github.com/thinking-machines-lab/tinker-cookbook/blob/main/tinker_cookbook/hyperparam_utils.py#L66
